File: assignments/haskell/triangle/triangle_test.hs
import Test.HUnit (Assertion, (@=?), runTestTT, Test(..), Counts(..))
import System.Exit (ExitCode(..), exitWith)
import Triangle (TriangleType(..), triangleType)

exitProperly :: IO Counts -> IO ()
exitProperly m = do
  counts <- m
  exitWith $ if failures counts /= 0 || errors counts /= 0 then ExitFailure 1 else ExitSuccess

testCase :: String -> Assertion -> Test
testCase label assertion = TestLabel label (TestCase assertion)

main :: IO ()
main = exitProperly $ runTestTT $ TestList
       [ TestList triangleTests ]

tri :: Int -> Int -> Int -> TriangleType
tri = triangleType

triangleTests :: [Test]
triangleTests = map TestCase
  [ Equilateral @=? tri 2 2 2
  , Equilateral @=? tri 10 10 10
  , Isosceles @=? tri 3 4 4
  , Isosceles @=? tri 4 3 4
  , Scalene @=? tri 3 4 5
  , Illogical @=? tri 1 1 50
  , Illogical @=? tri 1 2 1
  ]

File: assignments/clojure/triangle/triangle_test.clj
(ns triangle.test (:require [clojure.test :refer :all]))
(load-file "triangle.clj")

(deftest equilateral-1
  (is (= :equilateral (triangle 2 2 2))))
(deftest equilateral-2
  (is (= :equilateral (triangle 10 10 10))))
(deftest isoceles-1
  (is (= :isosceles (triangle 3 4 4))))
(deftest isoceles-2
  (is (= :isosceles (triangle 4 3 4))))
(deftest scalene
  (is (= :scalene (triangle 3 4 5))))
(deftest invalid-1
  (is (= :illogical (triangle 1 1 50))))
(deftest invalid-2
  (is (= :illogical (triangle 1 2 1))))

(run-tests)

